### PR TITLE
14 create role and permissions entities

### PR DESCRIPTION
**Pull request by AlbinNils98**
Created role and permission entities and repositories. I also linked them to the user entity. 

The data and schema sql files are from pr branch #39.

### CODE REVIEW

**Comment by AlbinNils98** [comment]
Made some changes in the last commit to data and schema sql files. I changed all tables to singular and added truncate table to make a clean slate everytime you start the application (for development purposes).

File: src/main/java/org/fungover/system2024/entities/User.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Size (max = 255)
+    @NotNull
+    @Column(name = "email", nullable = false)
+    private String email;
+
+    @Size (max = 255)
+    @NotNull
+    @Column(name = "password", nullable = false, unique = true)
+    private String password;
```
[the remark this answers is not in the document]
Comment (MikaelaMathisson) [reply src/main/java/org/fungover/system2024/entities/User.java:31]: Maybe look into this? :)

File: src/main/java/org/fungover/system2024/entities/User.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private String name;
+
+    @Size (max = 255)
+    @NotNull
+    @Column(name = "email", nullable = false)
+    private String email;
+
+    @Size (max = 255)
+    @NotNull
+    @Column(name = "password", nullable = false, unique = true)
```
[the remark this answers is not in the document]
Comment (MikaelaMathisson) [reply src/main/java/org/fungover/system2024/entities/User.java:30]: This also looks like something to look into. To optimize database storage?

**Review by MikaelaMathisson — DISMISSED** [review]
The code looks good and clean. I have made some comments for things that could be change, that the bot gave ideas on. Its not necessary things but would maybe be good to look into.
The code does what it should and has good names. I liked that you change everything from example roles to role, it makes it easier.
There's no tests for this code, but i will be doing them later on when this PR is approved and merged, so no worries 👍 Great work!

**Comment by MikaelaMathisson** [comment]
I saw now, 20 mins after my approve, that theres conflicts that must be resolved!

**Comment by AlbinNils98** [comment]
Resolved the merge conflict with main. Minor conflicts with merged pr #12. Awaiting reviews!

**Review by MikaelaMathisson — APPROVED** [review]
The code looks good and clean.
The code does what it should and has good clear simple names. I liked that you changed everything from example roles to role, it makes it easier.
There's no tests for this code, but i will be doing them later on when this PR is approved and merged, so no worries 👍 Great work! And great job working through and fixing the conflicts after my message.  Approving!

**Comment by AlbinNils98** [comment]
> This looks good Albin! The only thing I'm thinking about is the structure of the files (entities) in the project. You have put it in the entities folder but @sebastianpiresmolin has put the file in a separate folder? Maybe this is something we have to organize later.

This is definetly true, but when i worked with my issue, @sebastianpiresmolin 's wasn't merged yet and i didn't know how he had structured it. I agree we shouldn't have all entities in a specific folder so a valid package name could maybe be User?

**Pull request merged**